mobile/src/src/components/Housing/Listings/ListingsContainer.jsx

```diff
@@ -1,79 +1,72 @@
 import React, { Component } from "react";
 import { StyleSheet, View, FlatList } from "react-native";
-import ListingView from "./ListingView"
+import ListingView from "./ListingView";
 import { ListingsEndpoint } from "../../../utils/endpoints";
 
-class ListingContainer extends Component{
+class ListingContainer extends Component {
+  state = {
+    listings: [],
+    page: 1,
+    filters: {},
+    nextPageExists: true,
+  };
 
-    state = {
-        listings: [],
-        page: 1,
-        hasMoreListings: true
-    };
+  constructor(props) {
+    super(props);
+    this.getListings();
+  }
+
+  getListings = () => {
+    ListingsEndpoint.list(
+      this.state.page,
+      this.state.filters, //TODO add filter functionality and proper failure handling
+      (response) => {
+        const nextPageExists = response.data.next !== null;
+        let seen = {};
 
-    extendListings = (newListings) => {
         this.setState({
-                          listings: this.state.listings.concat(newListings),
+          listings: this.state.listings
+            .concat(response.data.results)
+            .filter((t) =>
+              seen.hasOwnProperty(t.id) ? false : (seen[t.id] = true)
+            ),
+          page: nextPageExists ? this.state.page + 1 : this.state.page,
```

Comment:
If I understand this correctly, when the user is on page 1, the `page` value will be 2, when they're on page 2, it'll be 3, etc.  If that's the case, we might want some extra clarity, so in that case maybe rename `page` to `nextPage` or smth.

Reply:
@shashankanandkannan true